MapleServer2/PacketHandlers/Game/PremiumClubHandler.cs

```diff
@@ -64,12 +84,50 @@ private static void HandleOpenPurchaseWindow(GameSession session)
 
         private static void HandlePurchaseMembership(GameSession session, PacketReader packet)
         {
-            int packageId = packet.ReadInt();
-            session.Send(PremiumClubPacket.PurchaseMembership(packageId));
-            // TODO grab data from \table\vipgoodstable.xml for pricing, buff duration, and bonus items
-            long expiration = 2592847227; // temporarilyy plugging in expiration time
-            session.Send(PremiumClubPacket.ActivatePremium(session.FieldPlayer, expiration));
-            session.Player.IsVIP = true;
+            int vipId = packet.ReadInt();
+
+            if (!PremiumClubPackageMetadataStorage.IsValid(vipId))
+            {
+                return;
+            }
+
+            PremiumClubPackageMetadata vipPackage = PremiumClubPackageMetadataStorage.GetMetadata(vipId);
+
+            if (!session.Player.Wallet.Meret.Modify(vipPackage.Price))
```

Comment:
If it's possible to buy VIP with any type of Meret you need to use `Wallet.RemoveMerets()`